frontend/javascripts/oxalis/model/sagas/saga_helpers.ts

```diff
@@ -113,4 +113,36 @@ export function* ensureMaybeActiveMappingIsLocked(
   }
 }
 
+export function* requestBucketModificationInVolumeTracing(
+  volumeTracing: VolumeTracing,
+): Saga<boolean> {
+  /*
+   * Should be called when the modification of bucket data is about to happen. If
+   * the saga returns false, the modification should be cancelled (this happens if
+   * the user is not okay with locking the mapping).
+   *
+   * In detail: When the bucket data of a volume tracing is supposed to be mutated, we need to do
+   * two things:
+   * 1) ensure that the current mapping (or no mapping) is locked so that the mapping is not
+   *    changed later (this would lead to inconsistent data). If the mapping state is not yet
+   *    locked, the user is asked whether it is okay to lock it.
+   *    If the user confirms this, the mapping is locked and we can continue with (2). If the
+   *    user denies the locking request, the original bucket mutation will NOT be executed.
+   * 2) volumeTracing.volumeBucketDataHasChanged will bet set to true if the user didn't
+   *    deny the request for locking the mapping.
+   */
+
+  const { isMappingLockedIfNeeded } = yield* call(ensureMaybeActiveMappingIsLocked, volumeTracing);
+  if (!isMappingLockedIfNeeded) {
+    return false;
+  }
+
+  // Mark that bucket data has changed
+  yield* put({
+    type: "SET_VOLUME_BUCKET_DATA_HAS_CHANGED",
+    tracingId: volumeTracing.tracingId,
+  });
```

Comment:
I'd use the constructor for the action here:
Moreover, the action states, that the member of the action should be the `layerName` and not the `tracingId` -> this is kinda confusing. Does this even work?
Something like this might be more correct
```suggestion
  yield* put(setVolumeBucketDataHasChangedAction(volumeTracing.tracingId));
```

Reply:
> I'd use the constructor for the action here

whoops, this is what I get for using windsurf to code this :see_no_evil:

> Moreover, the action states, that the member of the action should be the layerName and not the tracingId -> this is kinda confusing. Does this even work?

for volume tracings, `layerName` is always identical to the `tracingId`. it's a bit confusing and I'd love to change this, but the current implementation is in line with the rest. the reason for this is that there are some actions that both work on volumetracing layers as well as segmentation layer (these don't have tracing ids). so, the layer name is the common denominator.

Reply:
But this action is only designed to be called on volumetracings, isn't it?  Because in that case `tracingId` would also be correct / more precise

Reply:
ok, you are right :) for this specific case, tracingId also works without problems. I changed it now!